scrapers_next/al/people.py

```diff
@@ -0,0 +1,144 @@
+import re
+import attr
+from spatula import HtmlListPage, HtmlPage, XPath, CSS
+from openstates.models import ScrapePerson
+
+
+@attr.s(auto_attribs=True)
+class PartialMember:
+    url: str
+    chamber: str = ""
+
+
+class LegDetail(HtmlPage):
+    example_source = "http://www.legislature.state.al.us/aliswww/ISD/ALRepresentative.aspx?NAME=Alexander&OID_SPONSOR=100537&OID_PERSON=7710&SESSNAME=Regular%20Session%202022"
+
+    def process_page(self):
+
+        name = (
+            CSS(".container-main #ContentPlaceHolder1_lblMember")
+            .match_one(self.root)
+            .text_content()
+        )
+
+        if self.input.chamber == "upper":
+            name_split = re.split("SENATOR|, ", name)
+        elif self.input.chamber == "lower":
+            name_split = re.split("REPRESENTATIVE|, ", name)
+        full_name = name_split[2] + name_split[1]
+
+        table = CSS("#ContentPlaceHolder1_TabSenator_TabLeg_gvLEG").match_one(self.root)
+
+        for tr in CSS("tr").match(table):
+            type, info = CSS("td").match(tr)
+            type = type.text_content()
+            info = info.text_content()
+
+            if type == "Affiliation:":
+                party = ""
+                if info == "(R)":
+                    party = "Republican"
+                elif info == "(D)":
+                    party = "Democrat"
+                else:
+                    party = info
+                    if party == "( )":
+                        self.logger.warning("this seat may be vacant")
+                        # will assign it a random party for scraping purposes, but the warning should be addressed:
+                        party = "Democrat"
+
+            elif type == "District:":
+                district = info.split(" ")[2]
+            elif type == "County:":
+                county = info
+            elif type == "Phone Number:":
+                phone = info
+            elif type == "Fax Number:":
```

Comment:
if any of these conditionals are never met in a loop, the fax/etc. from the prior iteration will be used

at the top of the for loop it'd be good to initialize all of them to None/empty strings accordingly, so that no information can bleed over from the prior iteration